vm.go

```diff
@@ -186,6 +186,9 @@ func floatToInt(f float64) (result int64, ok bool) {
 }
 
 func floatToValue(f float64) (result Value) {
+	if i, ok := floatToInt(f); ok {
```

Comment:
does this pass all the tests in baas? an easy way to run this locally is to replace the `replace github.com/dop251/goja => github.com/mongodb-forks/goja v1.1.8` line in the `go.mod` file in baas to `replace github.com/dop251/goja => {YOUR_LOCAL_WORKING_DIRECTORY}/github.com/mongodb-forks`

Reply:
Local baas evergreen tests passed with the change, should be good to go now